op-node/rollup/derive/kromampt_upgrade_transactions.go

```diff
@@ -0,0 +1,300 @@
+	},
+}
+
+// ChainConfig constructor arguments mapping by ChainID
+type ChainConfig struct {
+	protocolVaultRecipient common.Address
+	l1FeeVaultRecipient    common.Address
+}
+
+// DeploymentData deployment bytecode information including constructor arguments
+type DeploymentData struct {
+	sequencerFeeVaultDeploymentBytecodeWithArg []byte
+	baseFeeVaultDeploymentBytecodeWithArg      []byte
+	l1FeeVaultDeploymentBytecodeWithArg        []byte
+}
+
+// getDeploymentData retrieves data that is necessary for deployments of predeploy contracts
+// at the previous block of Kroma MPT upgrade.
+func getDeploymentData(chainID *big.Int) (*DeploymentData, error) {
+	chainConfig, exists := ChainConfigs[chainID.String()]
+	if !exists {
+		chainConfig = ChainConfigs[strconv.Itoa(KromaLocalDevnetChainID)] // Default to local devnet configuration
+	}
+
+	sequencerBytecode, err := createDeploymentBytecode(sequencerFeeVaultDeploymentBytecode, chainConfig.protocolVaultRecipient)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create sequencerFeeVault deployment bytecode: %w", err)
+	}
+
+	baseFeeBytecode, err := createDeploymentBytecode(baseFeeVaultDeploymentBytecode, chainConfig.protocolVaultRecipient)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create baseFeeVault deployment bytecode: %w", err)
+	}
+
+	l1FeeBytecode, err := createDeploymentBytecode(l1FeeVaultDeploymentBytecode, chainConfig.l1FeeVaultRecipient)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create l1FeeVault deployment bytecode: %w", err)
+	}
+
+	return &DeploymentData{
+		sequencerFeeVaultDeploymentBytecodeWithArg: sequencerBytecode,
+		baseFeeVaultDeploymentBytecodeWithArg:      baseFeeBytecode,
+		l1FeeVaultDeploymentBytecodeWithArg:        l1FeeBytecode,
+	}, nil
+}
+
+// createDeploymentBytecode returns the result, including information about the constructor arguments
+// in the deploymentByteCode.
+func createDeploymentBytecode(bytecode []byte, args ...interface{}) ([]byte, error) {
+	encodedArgs, err := encodeConstructorArg(args...)
+	if err != nil {
+		return nil, err
+	}
+	// combine bytecode with encoded arguments
+	deploymentBytecode := append(bytecode, encodedArgs...)
+	return deploymentBytecode, nil
+}
+
+func encodeConstructorArg(args ...interface{}) ([]byte, error) {
+	// define ABI arguments dynamically based on the types of args
+	constructorArgs := abi.Arguments{}
+
+	for _, arg := range args {
+		var argType abi.Type
+		var err error
+
+		switch v := arg.(type) {
+		case common.Address:
+			argType, err = abi.NewType("address", "", nil)
+		case uint64:
+			argType, err = abi.NewType("uint64", "", nil)
+		case *big.Int:
+			argType, err = abi.NewType("uint256", "", nil)
+		case []byte:
+			argType, err = abi.NewType("bytes", "", nil)
+		case bool:
+			argType, err = abi.NewType("bool", "", nil)
+		default:
+			return nil, fmt.Errorf("unsupported argument type: %T", v)
```

Comment:
Why not restrict only the expected types?
```suggestion
		default:
			return nil, fmt.Errorf("unexpected argument type: %T", v)
```